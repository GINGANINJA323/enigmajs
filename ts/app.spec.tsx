/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import { render } from '@testing-library/react';
import App from './app';

describe('App', () => {
  test('renders correctly', () => {
    const result = render(<App />)

    expect(result).toMatchSnapshot();
  });
});