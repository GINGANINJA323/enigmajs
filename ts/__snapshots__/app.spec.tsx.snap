// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fEOsli eDwaar"
      >
        <div
          class="sc-ksZaOG frIDmL"
        >
          <div
            class="sc-hAZoDl eMkmPI"
          >
            <h1
              class="sc-dkzDqf jnhqor"
            >
              EnigmaJS: WWII Encryption
            </h1>
          </div>
          <div
            class="sc-fnykZs hoeozf"
          >
            <div
              class="sc-jSMfEi iomruJ"
            >
              <button
                class="sc-bczRLJ sc-gKXOVf bLqGOH hTYvza"
              >
                About
              </button>
              <button
                class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
              >
                Text Entry
              </button>
              <button
                class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
              >
                Plugboard
              </button>
              <button
                class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
              >
                Rotors
              </button>
              <button
                class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
              >
                Reflector Mode
              </button>
            </div>
            <div
              class="sc-breuTD gXhIYN"
            >
              <p>
                This is an Enigma simulator in JavaScript, with a ReactJS front-end. Use the navigation options above to continue.
              </p>
              <p>
                The Enigma machine was an encryption system used extensively by the German forces during World War 2. It works using 3 main components: the Plugboard, the Rotors and the Reflectors. The plugboard performs basic letter substitution using wires to connect a pair of letters together. The rotors perform a similar function, except they spin on different intervals as the message is typed out. The rotors are used twice per keypress with the reflector being used to perform a substitution in between. For more information, see 
                <a
                  href="https://www.cryptomuseum.com/crypto/enigma/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  here
                </a>
                . The code for this website, and the underlying Enigma logic, can be found on the Git repo 
                <a
                  href="https://github.com/GINGANINJA323/enigmajs"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  here
                </a>
                .
              </p>
            </div>
          </div>
          <div
            class="Toastify"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fEOsli eDwaar"
    >
      <div
        class="sc-ksZaOG frIDmL"
      >
        <div
          class="sc-hAZoDl eMkmPI"
        >
          <h1
            class="sc-dkzDqf jnhqor"
          >
            EnigmaJS: WWII Encryption
          </h1>
        </div>
        <div
          class="sc-fnykZs hoeozf"
        >
          <div
            class="sc-jSMfEi iomruJ"
          >
            <button
              class="sc-bczRLJ sc-gKXOVf bLqGOH hTYvza"
            >
              About
            </button>
            <button
              class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
            >
              Text Entry
            </button>
            <button
              class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
            >
              Plugboard
            </button>
            <button
              class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
            >
              Rotors
            </button>
            <button
              class="sc-bczRLJ sc-gKXOVf bLqGOH jXEIIj"
            >
              Reflector Mode
            </button>
          </div>
          <div
            class="sc-breuTD gXhIYN"
          >
            <p>
              This is an Enigma simulator in JavaScript, with a ReactJS front-end. Use the navigation options above to continue.
            </p>
            <p>
              The Enigma machine was an encryption system used extensively by the German forces during World War 2. It works using 3 main components: the Plugboard, the Rotors and the Reflectors. The plugboard performs basic letter substitution using wires to connect a pair of letters together. The rotors perform a similar function, except they spin on different intervals as the message is typed out. The rotors are used twice per keypress with the reflector being used to perform a substitution in between. For more information, see 
              <a
                href="https://www.cryptomuseum.com/crypto/enigma/"
                rel="noopener noreferrer"
                target="_blank"
              >
                here
              </a>
              . The code for this website, and the underlying Enigma logic, can be found on the Git repo 
              <a
                href="https://github.com/GINGANINJA323/enigmajs"
                rel="noopener noreferrer"
                target="_blank"
              >
                here
              </a>
              .
            </p>
          </div>
        </div>
        <div
          class="Toastify"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
